### PR TITLE
Take

### PR DESCRIPTION
**Pull request by snreynolds**
## Related Issue
Which issue does this pull request resolve?

## Description of changes

### CODE REVIEW

File: src/PositionManager.sol [hunk 1]
```diff
@@ -186,6 +186,9 @@ contract PositionManager is
         } else if (action == Actions.SWEEP) {
             (Currency currency, address to) = params.decodeCurrencyAndAddress();
             _sweep(currency, _mapRecipient(to));
+        } else if (action == Actions.TAKE) {
+            (Currency currency, address recipient, uint256 amount) = params.decodeCurrencyAddressAndUint256();
+            _take(currency, _mapRecipient(recipient), _mapTakeAmount(amount, currency));
```
Comment (saucepoint) [inline src/PositionManager.sol:191]: hmm `params.decodeCurrencyAddressAndUint256()` isnt tested in `CalldataDecoder.t.sol` 🧐 

can you add?

File: test/position-managers/PositionManager.t.sol [hunk 2]
```diff
@@ -878,5 +878,74 @@ contract PositionManagerTest is Test, PosmTestSetup, LiquidityFuzzers {
...
+            Actions.DECREASE_LIQUIDITY,
+            abi.encode(tokenId, config, 1e18, MIN_SLIPPAGE_DECREASE, MIN_SLIPPAGE_DECREASE, ZERO_BYTES)
+        );
+        plan.add(
+            Actions.MINT_POSITION,
+            abi.encode(configMint, 1e18, MAX_SLIPPAGE_INCREASE, MAX_SLIPPAGE_INCREASE, Constants.MSG_SENDER, ZERO_BYTES)
+        );
+        plan.add(Actions.TAKE, abi.encode(key.currency0, Constants.MSG_SENDER, Constants.OPEN_DELTA));
+        plan.add(Actions.SETTLE, abi.encode(key.currency1, Constants.OPEN_DELTA, true));
+        bytes memory calls = plan.finalizeModifyLiquidityWithTake(config.poolKey, Constants.MSG_SENDER);
```
Comment (saucepoint) [inline test/position-managers/PositionManager.t.sol:936]: should be in Execute.t.sol, but can save for a later issue when we reorg tests

File: test/position-managers/PositionManager.t.sol [hunk 3]
```diff
@@ -878,5 +878,74 @@ contract PositionManagerTest is Test, PosmTestSetup, LiquidityFuzzers {
...
+
+        lpm.modifyLiquidities(calls, _deadline);
+        BalanceDelta deltaDecrease = hook.deltas(0);
+        BalanceDelta deltaMint = hook.deltas(1);
+
+        assertEq(deltaMint.amount0(), 0); // there is no currency0 in the new position
+        assertEq(currency0.balanceOfSelf(), balanceBefore0 + uint256(int256(deltaDecrease.amount0())));
+        assertEq(
+            currency1.balanceOfSelf(), balanceBefore1 - uint256(-int256(deltaDecrease.amount1() + deltaMint.amount1()))
+        );
```
Comment (saucepoint) [inline test/position-managers/PositionManager.t.sol:946]: can we also assert `currency1.balanceOfSelf() < balanceBefore1` to confirm that we paid additional tokens with settle

maybe also `deltaDecrease.amount1() < -deltaMint.amount1()`

File: test/position-managers/PositionManager.t.sol
Comment (saucepoint) [inline test/position-managers/PositionManager.t.sol:1]: can we test:

1. DECREASE
2. TAKE portion of amount0 to recipient
3. TAKE portion of amount1 to recipient
4. TAKE OPEN_DELTAS to owner/msgSender
5. TAKE OPEN_DELTAS to owner/msgSender

Comment (saucepoint) [reply test/position-managers/PositionManager.t.sol:1]: (will create a separate issue for TAKE / TAKE_PAIR to posm)

File: test/position-managers/PositionManager.t.sol (hunk 2 above)
Comment (saucepoint) [reply test/position-managers/PositionManager.t.sol:936]: super neat test tho

File: test/position-managers/PositionManager.t.sol
Comment (snreynolds) [reply test/position-managers/PositionManager.t.sol:1]: oo yep good test

File: src/PositionManager.sol (hunk 1 above)
Comment (snreynolds) [reply src/PositionManager.sol:191]: added

File: test/position-managers/PositionManager.t.sol (hunk 2 above)
Comment (snreynolds) [reply test/position-managers/PositionManager.t.sol:936]: will move later

File: test/position-managers/PositionManager.t.sol
Comment (snreynolds) [reply test/position-managers/PositionManager.t.sol:1]: https://github.com/Uniswap/v4-periphery/issues/258

**Review by saucepoint — APPROVED** [review]

**Pull request merged**